packages/ra-ui-materialui/src/field/FunctionField.tsx

```diff
@@ -48,9 +48,13 @@ FunctionField.propTypes = {
     render: PropTypes.func.isRequired,
 };
 
+export type FunctionFieldRenderer<
+    RecordType extends Record<string, unknown> = Record<string, any>
+> = (record: RecordType, source?: string) => React.ReactNode;
```

Comment:
I was unsure on what to name this. It may be helpful to further abstract this into a type that’s shared by `<WithRecord>`